src/m365/spo/commands/serviceprincipal/serviceprincipal-permissionrequest-list.ts

```diff
@@ -58,6 +71,40 @@ class SpoServicePrincipalPermissionRequestListCommand extends SpoCommand {
       this.handleRejectedPromise(err);
     }
   }
+
+  private async getOAuth2PermissionGrants(spoClientExtensibilityWebApplicationPrincipalId: string): Promise<string[] | null> {
+    const requestOptions: CliRequestOptions = {
+      url: `https://graph.microsoft.com/v1.0/oAuth2Permissiongrants/?$filter=clientId eq '${spoClientExtensibilityWebApplicationPrincipalId}' and consentType eq 'AllPrincipals'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    const response: { value: OAuth2PermissionGrant[] } = await request.get<{ value: OAuth2PermissionGrant[] }>(requestOptions);
+    if (response.value && response.value.length > 0) {
+      return response.value[0].scope!.split(' ');
+    }
+
+    return null;
+  }
+
+  private async getSPOClientExtensibilityWebApplicationPrincipalId(): Promise<string | null> {
+    const requestOptions: CliRequestOptions = {
+      url: `https://graph.microsoft.com/v1.0/servicePrincipals/?$filter=displayName eq 'SharePoint Online Client Extensibility Web Application Principal'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    const response: { value: ServicePrincipal[] } = await request.get(requestOptions);
```

Comment:
Here, we can use `ODataResponse<ServicePrincipal>` as the get call's return value